trace.go

```diff
@@ -174,6 +174,9 @@ func GetFields(err error) map[string]interface{} {
 
 // WrapWithMessage wraps the original error into Error and adds user message if any
 func WrapWithMessage(err error, message interface{}, args ...interface{}) Error {
+	if err == nil {
```

Comment:
perhaps implement this function in terms of `Wrap()`?

    func WrapWithMessage(err error, message interface{}, args ...interface{}) Error {
        return Wrap(err, append([]interface{}{message}, args...)...)
    }

( `Wrap(err, message, args...)` unfortunately is not allowed by Go - the `...` actual arg needs to exactly match the `...` formal arg, hence the shenanigans with `append`)